grpc-api/src/main/java/io/deephaven/grpc_api/arrow/ArrowFlightUtil.java

```diff
@@ -301,7 +295,7 @@ public void onCompleted() {
                     // transfer ownership to submit's liveness scope, drop our extra reference
                     resultTable.manageWithCurrentScope();
                     resultTable.dropReference();
-                    GrpcUtil.safelyExecute(observer::onCompleted);
+                    GrpcUtil.safelyExecuteLocked(observer, observer::onCompleted);
```

Comment:
Hmm. I forgot about this method. We should have `onErrorSynchronized` and `safelyExecuteLocked` re-use the same suffix. I don't care which; but `safelyExecuteSynchronized` is probably more clear.

Reply:
This one I doubt more than the others... should we also make this force a lock? Most of these are very short and involve the observer (being in GrpcUtils and all), thoughts?